spinalcordtoolbox/scripts/sct_compute_hausdorff_distance.py

```diff
@@ -91,7 +91,7 @@ def transitions(self, neighbours):
         """
         # now = time.time()
         n = neighbours + neighbours[0:1]      # P2, P3, ... , P8, P9, P2
-        s = np.sum((n1, n2) == (0, 1) for n1, n2 in zip(n, n[1:]))  # (P2,P3), (P3,P4), ... , (P8,P9), (P9,P2)
+        s = np.sum(np.fromiter([(n1, n2) == (0, 1) for n1, n2 in zip(n, n[1:])], dtype=bool))  # (P2,P3), (P3,P4), ... , (P8,P9), (P9,P2)
```

Comment:
I think using Python's builtin `sum` (which treats `True` as 1 and `False` as 0) is the simpler fix here:
```suggestion
        s = sum((n1, n2) == (0, 1) for n1, n2 in zip(n, n[1:]))  # (P2,P3), (P3,P4), ... , (P8,P9), (P9,P2)
```
(Besides, adding _both_ `np.fromiter` and square brackets is overkill.)

Reply:
I very much agree with the sum change! 

But, a note about `np.fromiter` + square brackets: I tried to get rid of the square brackets at first, but ran into:

![image](https://user-images.githubusercontent.com/16181459/233392312-46bbd52d-9283-46e1-8e02-b06c73733f5f.png)

```
E     File "/home/joshua/repos/spinalcordtoolbox/spinalcordtoolbox/scripts/sct_compute_hausdorff_distance.py", line 94
E       s = np.sum(np.fromiter((n1, n2) == (0, 1) for n1, n2 in zip(n, n[1:]), dtype=bool))  # (P2,P3), (P3,P4), ... , (P8,P9), (P9,P2)
E                              ^
E   SyntaxError: Generator expression must be parenthesized
```

This was news to me! :stuck_out_tongue_closed_eyes:

Reply:
Ah, that's true! But still, parenthesizing with `[...]` turns it into a bona fide list, whereas parenthesizing with `(...)` as the `SyntaxError` suggests keeps it as a generator.

Reply:
Aha!! Thank you, I didn't realize. 

I presumed that using `(...)` would turn it into a tuple (i.e. there wasn't a significant difference between `[...]` and `(...)`). But, TIL that [there is no "tuple comprehension"](https://stackoverflow.com/q/16940293), and that `(...)` indeed returns a generator instead.